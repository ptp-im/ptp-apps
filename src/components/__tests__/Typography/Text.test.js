import * as React from 'react';

import { render } from '@testing-library/react-native';
import renderer from 'react-test-renderer';

import { tokens } from '../../../styles/themes/v3/tokens';
import Text from '../../Typography/Text.tsx';

const content = 'Something rendered as a child content';

it('renders every variant of Text with children as content', () => {
  const variants = (
    <>
      <Text variant='displayLarge'>{content}</Text>
      <Text variant='displayMedium'>{content}</Text>
      <Text variant='displaySmall'>{content}</Text>

      <Text variant='headlineLarge'>{content}</Text>
      <Text variant='headlineMedium'>{content}</Text>
      <Text variant='headlineSmall'>{content}</Text>

      <Text variant='titleLarge'>{content}</Text>
      <Text variant='titleMedium'>{content}</Text>
      <Text variant='titleSmall'>{content}</Text>

      <Text variant='bodyLarge'>{content}</Text>
      <Text variant='bodyMedium'>{content}</Text>
      <Text variant='bodySmall'>{content}</Text>

      <Text variant='labelLarge'>{content}</Text>
      <Text variant='labelMedium'>{content}</Text>
      <Text variant='labelSmall'>{content}</Text>
    </>
  );

  const tree = renderer.create(variants).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders v3 Text component without variant with default fontWeight and fontFamily', () => {
  const { getByTestId } = render(
    <Text testID='text-without-variant'>{content}</Text>
  );
  const { brandRegular, weightRegular } = tokens.md.ref.typeface;

  expect(getByTestId('text-without-variant')).toHaveStyle({
    fontFamily: brandRegular,
    fontWeight: weightRegular,
  });
});
