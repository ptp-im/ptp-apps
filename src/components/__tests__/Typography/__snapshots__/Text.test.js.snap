// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders every variant of Text with children as content 1`] = `
Array [
  <Text
    style={
      Array [
        Object {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontSize": 57,
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 64,
        },
        Object {
          "textAlign": "left",
        },
        Object {
          "writingDirection": "ltr",
        },
        undefined,
      ]
    }
  >
    Something rendered as a child content
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontSize": 45,
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 52,
        },
        Object {
          "textAlign": "left",
        },
        Object {
          "writingDirection": "ltr",
        },
        undefined,
      ]
    }
  >
    Something rendered as a child content
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontSize": 36,
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 44,
        },
        Object {
          "textAlign": "left",
        },
        Object {
          "writingDirection": "ltr",
        },
        undefined,
      ]
    }
  >
    Something rendered as a child content
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontSize": 32,
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 40,
        },
        Object {
          "textAlign": "left",
        },
        Object {
          "writingDirection": "ltr",
        },
        undefined,
      ]
    }
  >
    Something rendered as a child content
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontSize": 28,
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 36,
        },
        Object {
          "textAlign": "left",
        },
        Object {
          "writingDirection": "ltr",
        },
        undefined,
      ]
    }
  >
    Something rendered as a child content
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontSize": 24,
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 32,
        },
        Object {
          "textAlign": "left",
        },
        Object {
          "writingDirection": "ltr",
        },
        undefined,
      ]
    }
  >
    Something rendered as a child content
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontSize": 22,
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 28,
        },
        Object {
          "textAlign": "left",
        },
        Object {
          "writingDirection": "ltr",
        },
        undefined,
      ]
    }
  >
    Something rendered as a child content
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontSize": 16,
          "fontWeight": "500",
          "letterSpacing": 0.15,
          "lineHeight": 24,
        },
        Object {
          "textAlign": "left",
        },
        Object {
          "writingDirection": "ltr",
        },
        undefined,
      ]
    }
  >
    Something rendered as a child content
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontSize": 14,
          "fontWeight": "500",
          "letterSpacing": 0.1,
          "lineHeight": 20,
        },
        Object {
          "textAlign": "left",
        },
        Object {
          "writingDirection": "ltr",
        },
        undefined,
      ]
    }
  >
    Something rendered as a child content
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontSize": 16,
          "fontWeight": "400",
          "letterSpacing": 0.15,
          "lineHeight": 24,
        },
        Object {
          "textAlign": "left",
        },
        Object {
          "writingDirection": "ltr",
        },
        undefined,
      ]
    }
  >
    Something rendered as a child content
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontSize": 14,
          "fontWeight": "400",
          "letterSpacing": 0.25,
          "lineHeight": 20,
        },
        Object {
          "textAlign": "left",
        },
        Object {
          "writingDirection": "ltr",
        },
        undefined,
      ]
    }
  >
    Something rendered as a child content
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontSize": 12,
          "fontWeight": "400",
          "letterSpacing": 0.4,
          "lineHeight": 16,
        },
        Object {
          "textAlign": "left",
        },
        Object {
          "writingDirection": "ltr",
        },
        undefined,
      ]
    }
  >
    Something rendered as a child content
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontSize": 14,
          "fontWeight": "500",
          "letterSpacing": 0.1,
          "lineHeight": 20,
        },
        Object {
          "textAlign": "left",
        },
        Object {
          "writingDirection": "ltr",
        },
        undefined,
      ]
    }
  >
    Something rendered as a child content
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontSize": 12,
          "fontWeight": "500",
          "letterSpacing": 0.5,
          "lineHeight": 16,
        },
        Object {
          "textAlign": "left",
        },
        Object {
          "writingDirection": "ltr",
        },
        undefined,
      ]
    }
  >
    Something rendered as a child content
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontSize": 11,
          "fontWeight": "500",
          "letterSpacing": 0.5,
          "lineHeight": 16,
        },
        Object {
          "textAlign": "left",
        },
        Object {
          "writingDirection": "ltr",
        },
        undefined,
      ]
    }
  >
    Something rendered as a child content
  </Text>,
]
`;
